File: Solutions/airport.ts
const { exec } = require('child_process');
exec('networksetup -listallhardwareports', (err, stdout, stderr) => {
  if (err) {
    console.error(`exec error: ${err}`);
    return;
  }
  console.log(`stdout: ${stdout}`);
  console.error(`stderr: ${stderr}`);
});




exec('/System/Library/PrivateFrameworks/Apple80211.framework/Versions/Current/Resources/airport -s', (error, stdout, stderr) => {
  if (error) {
      console.error(`执行错误: ${error.message}`);
      return;
  }
  if (stderr) {
      console.error(`错误信息: ${stderr}`);
      return;
  }

  console.log(`输出信息: ${stdout}`);
  
  // 处理输出
  const wifiList = stdout.trim().split('\n').map(line => {
      const parts = line.split(/\s+/);
      return {
          SSID: parts[0], // Wi-Fi 名称
          BSSID: parts[1], // MAC 地址
          RSSI: parts[2],  // 信号强度
          CHANNEL: parts[3], // 信道
          HT: parts[4],    // HT 支持
          CC: parts[5],    // 国家代码
          SECURITY: parts[6] // 安全性
      };
  });

  console.log(wifiList);
});

File: Solutions/airport.py
import subprocess
import json

def get_wifi_networks():
    try:
        # 调用 airport 命令，列出可用 Wi-Fi 网络
        result = subprocess.run(
            ["/System/Library/PrivateFrameworks/Apple80211.framework/Versions/Current/Resources/airport", "-s"],
            capture_output=True, text=True
        )
        
        networks = []
        for line in result.stdout.splitlines()[1:]:  # 跳过第一行标题
            parts = line.split()
            if len(parts) >= 2:
                ssid = " ".join(parts[:-1])  # Wi-Fi 名称可能包含空格
                signal_strength = parts[-1]
                networks.append({"SSID": ssid, "Signal": signal_strength})
        
        # 输出 JSON 格式的 Wi-Fi 网络信息
        print(json.dumps(networks))

    except Exception as e:
        print(f"Error: {e}")

get_wifi_networks()
